[Quarkus main] Native integration test oidc-wiremock-providers fails due to config value conversion issues

### Describe the bug

We started to test failures of the `io.quarkus.it.oidc.providers.OidcSlackProviderIT.testSlackWellKnownProvide` native integration test. It fails with:

```
[INFO] Running io.quarkus.it.oidc.providers.OidcSlackProviderIT
2026-07-21 11:21:16,823 INFO  [org.eclipse.aether.internal.impl.filter.GroupIdRemoteRepositoryFilterSource] (main) Loaded 1 groupId for remote repository gradle
2026-07-21 11:21:19,263 INFO  [io.quarkus.test.oidc.server.OidcWiremockTestResource] (pool-3-thread-1) Keycloak started in mock mode: http://localhost:36601
Executing "/home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/integration-tests/oidc-wiremock-providers/target/quarkus-integration-test-oidc-wiremock-providers-999-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:${quarkus.http.test-port:8081} -Dquarkus.log.file.path=/home/sgehwolf/Documents/openjdk/quarkus/quarkus-source/integration-tests/oidc-wiremock-providers/target/quarkus.log -Dquarkus.log.file.enabled=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.profile=prod -Dkeycloak.url=http://localhost:36601/auth -Dsmallrye.jwt.sign.key.location=privateKey.jwk -Dquarkus.native.enable-reports=true -Dquarkus.native.container-build=false"
Configuration validation failed:
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.*.authentication.state-cookie-age with the config value "5M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.*.connection-timeout with the config value "10s" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.*.jwks.cache-time-to-live with the config value "10M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.*.logout.backchannel.token-cache-time-to-live with the config value "10M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.*.token.forced-jwk-refresh-interval with the config value "10M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.*.token.refresh-token-cache-time-to-live with the config value "0S" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.authentication.state-cookie-age with the config value "5M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.connection-timeout with the config value "10s" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.jwks.cache-time-to-live with the config value "10M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.logout.backchannel.token-cache-time-to-live with the config value "10M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.token.forced-jwk-refresh-interval with the config value "10M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.token.refresh-token-cache-time-to-live with the config value "0S" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
	java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.token-cache.time-to-live with the config value "3M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
Process exited with non-zero status: 1
2026-07-21 11:21:20,063 INFO  [io.quarkus.test.oidc.server.OidcWiremockTestResource] (main) Keycloak was shut down
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.282 s <<< FAILURE! -- in io.quarkus.it.oidc.providers.OidcSlackProviderIT
[ERROR] io.quarkus.it.oidc.providers.OidcSlackProviderIT.testSlackWellKnownProvider -- Time elapsed: 0.008 s <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Unable to successfully launch process '223058'. Exit code is: '1'.
	at io.quarkus.test.junit.QuarkusIntegrationTestExtension.throwBootFailureException(QuarkusIntegrationTestExtension.java:425)
	at io.quarkus.test.junit.QuarkusIntegrationTestExtension.beforeEach(QuarkusIntegrationTestExtension.java:130)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.RuntimeException: Unable to successfully launch process '223058'. Exit code is: '1'.
	at io.quarkus.test.common.LauncherUtil.ensureProcessIsAlive(LauncherUtil.java:131)
	at io.quarkus.test.common.LauncherUtil.waitForCapturedListeningData(LauncherUtil.java:94)
	at io.quarkus.test.common.DefaultNativeImageLauncher.start(DefaultNativeImageLauncher.java:100)
	at io.quarkus.test.junit.IntegrationTestUtil.startLauncher(IntegrationTestUtil.java:131)
	at io.quarkus.test.junit.QuarkusIntegrationTestExtension.doProcessStart(QuarkusIntegrationTestExtension.java:359)
	at io.quarkus.test.junit.QuarkusIntegrationTestExtension.ensureStarted(QuarkusIntegrationTestExtension.java:199)
	at io.quarkus.test.junit.QuarkusIntegrationTestExtension.beforeAll(QuarkusIntegrationTestExtension.java:154)
	... 1 more
```

Note all these messages:

```
java.lang.IllegalArgumentException: SRCFG00039: The config property quarkus.oidc.*.authentication.state-cookie-age with the config value "5M" threw an Exception whilst being converted SRCFG00020: Failed to convert value with static method
```

### Expected behavior

Native integration tests of `oidc-wiremock-providers` passes.

### Actual behavior

Native integration tests with a stock Mandrel-25.0.3.0-Final builder fails with conversion issues.

### How to Reproduce?

Reproducer:

1. Build quarkus main (revision 7c7bc9e63129c5c2e4c881f5d4c44321ec8c38a4 should do)
2. Run native integration test `oidc-wiremock-providers` with a Mandrel 25.0.3.0-Final builder (for example).
3. Observe the test failure.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

Java 17

### Mandrel or GraalVM version (if different from Java)

Mandrel 25.0.3.0-Final

### Quarkus version or git rev

7c7bc9e63129c5c2e4c881f5d4c44321ec8c38a4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

See also these CI runs where it fails:
https://github.com/graalvm/mandrel/actions/runs/29750010643/job/88388319066#step:12:13012
https://github.com/graalvm/mandrel/actions/runs/29791543242/job/88516026167#step:12:13141

--- a/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/steps/ConfigGenerationBuildStep.java
@@ -143,14 +143,27 @@ public class ConfigGenerationBuildStep {
                 "io.quarkus.runtime.configuration.RuntimeConfigBuilder$UuidConfigSource$Holder"));
     }
 
+    /**
+     * Registers all config components descriptors for native image.
+     * <p>
+     * The initialized Config used by Quarkus does not require them, but users may be instantiating their own Config.
+     */
     @BuildStep
     void nativeServiceProviders(BuildProducer<ServiceProviderBuildItem> providerProducer) throws IOException {
         providerProducer.produce(new ServiceProviderBuildItem(ConfigProviderResolver.class.getName(),
                 SmallRyeConfigProviderResolver.class.getName()));
         ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-        // TODO - radcortez - Move these services to be registered directly with the builder
         for (Class<?> serviceClass : Arrays.asList(
+                Converter.class,
+                ConfigSource.class,
+                ConfigSourceProvider.class,
+                ConfigSourceFactory.class,
+                ConfigSourceInterceptor.class,
+                ConfigSourceInterceptorFactory.class,
+                SecretKeysHandler.class,
+                SecretKeysHandlerFactory.class,
                 ConfigValidator.class,
+                SmallRyeConfigBuilderCustomizer.class,
                 ConfigMappingHandler.class)) {
             String serviceName = serviceClass.getName();
             Set<String> names = ServiceUtil.classNamesNamedIn(classLoader, SERVICES_PREFIX + serviceName);

